iso15118/shared/states.py

```diff
@@ -290,24 +311,34 @@ def create_next_message(
             except ValidationError as exc:
                 logger.exception(exc)
                 raise exc
+            self.message = to_be_exi_encoded
+        elif isinstance(next_msg, Base64):
+            # Incoming message is base64 encoded EXI message.
+            # So a base64 decode should retrieve the EXI stream.
+            self.message = next_msg
+            next_msg = base64.b64decode(next_msg.message)
         else:
             to_be_exi_encoded = next_msg
+            self.message = to_be_exi_encoded
 
-        self.next_msg = to_be_exi_encoded
-
-        # If either next_msg or next_msg_payload_type are None, the state's
+        # If either message or next_msg_payload_type are None, the state's
         # attribute next_v2gtp_msg will not be set. This causes the state
         # machine to raise a FaultyStateImplementationError if next state is
         # not set to Terminate, so no need to raise anything here.
         if next_msg and next_msg_payload_type:
```

Comment:
we actually dont need this next_msg check...we can use just the `to_be_exi_encoded`.
and we do:

```python
elif isinstance(next_msg, Base64):
            # Incoming message is base64 encoded EXI message.
            # So a base64 decode should retrieve the EXI stream.
            self.message = next_msg
            exi_payload = base64.b64decode(next_msg.message)
else:
    to_be_exi_encoded = next_msg
    self.message = to_be_exi_encoded

    # If either message or next_msg_payload_type are None, the state's
    # attribute next_v2gtp_msg will not be set. This causes the state
    # machine to raise a FaultyStateImplementationError if next state is
    # not set to Terminate, so no need to raise anything here.
    if to_be_exi_encoded and next_msg_payload_type:
          try:
              exi_payload = EXI().to_exi(to_be_exi_encoded, namespace)
          except EXIEncodingError as exc:
              logger.error(f"{exc}")
              self.next_state = Terminate
              raise
```

Reply:
I also would like then to rename the `next_msg` to `msg`...but maybe that is too much of a work for now...

Reply:
Done.